bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/tag/TagDTO.java

```diff
@@ -15,24 +15,33 @@
 import java.util.List;
 import java.util.Locale;
 
-import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.openhab.core.semantics.SemanticTags;
 import org.openhab.core.semantics.Tag;
+import org.openhab.core.semantics.TagInfo;
 
 /**
  * A DTO representing a Semantic {@link Tag}.
  *
  * @author Jimmy Tanagra - initial contribution
+ * @author Laurent Garnier - Add uid, description and editable
  */
-@NonNullByDefault
 public class TagDTO {
+    String uid;
     String name;
     String label;
+    String description;
     List<String> synonyms;
+    boolean editable;
 
-    public TagDTO(Class<? extends Tag> tag, Locale locale) {
+    public TagDTO() {
```

Comment:
I think this is only for de-serialization, correct? It can be protected or private.

Reply:
It will be necessary by caller of the REST API to provide tag data (add or update a custom tag).

Reply:
Removed